tflint/tflint.go

```diff
@@ -26,41 +38,83 @@ func RunTflintWithOpts(terragruntOptions *options.TerragruntOptions, terragruntC
+	if err != nil {
+		return err
+	}
+	variables = append(variables, tfVariables...)
+
 	terragruntOptions.Logger.Debugf("Initializing tflint in directory %s", terragruntOptions.WorkingDir)
 	cli, err := cmd.NewCLI(terragruntOptions.Writer, terragruntOptions.ErrWriter)
 	if err != nil {
 		return errors.WithStackTrace(err)
 	}
 
+	tflintArgs, externalTfLint := tflintArguments(hook.Execute[1:])
+
 	// tflint init
-	initArgs := []string{"tflint", "--init", "--config", configFile, terragruntOptions.WorkingDir}
-	statusCode := cli.Run(initArgs)
-	if statusCode != 0 {
-		return errors.WithStackTrace(ErrorRunningTflint{args: initArgs})
+	initArgs := []string{"tflint", "--init", "--config", configFile, "--chdir", terragruntOptions.WorkingDir}
+	if externalTfLint {
+		terragruntOptions.Logger.Debugf("Running external tflint init with args %v", initArgs)
+		_, err := shell.RunShellCommandWithOutput(terragruntOptions, terragruntOptions.WorkingDir, false, false, initArgs[0], initArgs[1:]...)
+		if err != nil {
+			return errors.WithStackTrace(ErrorRunningTflint{args: initArgs})
+		}
+	} else {
+		terragruntOptions.Logger.Debugf("Running internal tflint init with args %v", initArgs)
+		statusCode := cli.Run(initArgs)
+		if statusCode != 0 {
+			return errors.WithStackTrace(ErrorRunningTflint{args: initArgs})
+		}
 	}
 
 	// tflint execution
 	args := []string{"tflint"}
-	args = append(args, variables...)
 	args = append(args, "--config", configFile)
 	args = append(args, "--chdir", terragruntOptions.WorkingDir)
+	args = append(args, variables...)
+	args = append(args, tflintArgs...)
 
-	terragruntOptions.Logger.Debugf("Running tflint with args %v", args)
-	statusCode = cli.Run(args)
-
-	if statusCode == cmd.ExitCodeError {
-		return errors.WithStackTrace(ErrorRunningTflint{args: initArgs})
-	} else if statusCode == cmd.ExitCodeIssuesFound {
-		return errors.WithStackTrace(IssuesFound{})
-	} else if statusCode == cmd.ExitCodeOK {
+	if externalTfLint {
+		terragruntOptions.Logger.Debugf("Running external tflint with args %v", args)
+		_, err := shell.RunShellCommandWithOutput(terragruntOptions, terragruntOptions.WorkingDir, false, false, args[0], args[1:]...)
+		if err != nil {
+			return errors.WithStackTrace(ErrorRunningTflint{args: args})
+		}
 		terragruntOptions.Logger.Info("Tflint has run successfully. No issues found.")
 	} else {
-		return errors.WithStackTrace(UnknownError{statusCode: statusCode})
+		terragruntOptions.Logger.Debugf("Running internal tflint with args %v", args)
+		statusCode := cli.Run(args)
+
+		if statusCode == cmd.ExitCodeError {
+			return errors.WithStackTrace(ErrorRunningTflint{args: initArgs})
+		} else if statusCode == cmd.ExitCodeIssuesFound {
+			return errors.WithStackTrace(IssuesFound{})
+		} else if statusCode == cmd.ExitCodeOK {
+			terragruntOptions.Logger.Info("Tflint has run successfully. No issues found.")
+		} else {
+			return errors.WithStackTrace(UnknownError{statusCode: statusCode})
+		}
```

Comment:
```suggestion
		switch statusCode {
		case cmd.ExitCodeError:
			return errors.WithStackTrace(ErrorRunningTflint{args: initArgs})
		case cmd.ExitCodeIssuesFound:
			return errors.WithStackTrace(IssuesFound{})
		case cmd.ExitCodeOK:
			terragruntOptions.Logger.Info("Tflint has run successfully. No issues found.")
		default:
			return errors.WithStackTrace(UnknownError{statusCode: statusCode})
		}
```

Better readable code